### ISSUE TITLE
Host memory continues to be occupied

### ISSUE THREAD

**Issue by Laidawang** [lllyasviel/Fooocus#1646]
**Read Troubleshoot**

[x] I admit that I have read the [Troubleshoot](https://github.com/lllyasviel/Fooocus/blob/main/troubleshoot.md) before making this issue.

**Describe the problem**
The memory used by the program seems to be slowly increasing

**Full Console Log**
![image](https://github.com/lllyasviel/Fooocus/assets/85244566/6ba94a28-fe5a-44c4-97a6-8cf4c8ce71ec)
![image](https://github.com/lllyasviel/Fooocus/assets/85244566/dc3ad80e-056f-41da-9cfe-e14ab45cb1c4)
![image](https://github.com/lllyasviel/Fooocus/assets/85244566/a52f6b73-2edf-4d65-b746-5f566b2be41a)
This is a potential problem that cannot be expressed in log. You can see that the memory occupied by fooocus is slowly increasing.
My startup params: python main.py  --host 0.0.0.0 --all-in-fp16 --port 8000 --always-gpu

**Comment by mashb1t** [comment]
Thank you for the report, it sould be great if you could provide further information:
Does this happen repeatedly, so can you reproduce the behavior?
Is this only happening on your machine and what are you doing in Fooocus?

Can somebody else confirm this behavior and is experiencing also an increase in memory over time?

**Comment by Laidawang** [comment]
[memory.log](https://github.com/lllyasviel/Fooocus/files/13800296/memory.log)
I recorded the memory of 20 times outpainting(four directions). Memory usage is indeed rising, although it is not obvious
`#!/bin/bash

while true; do
  memory=$(ps aux | grep PID | awk '{print $6}')
  datetime=$(date "+%Y-%m-%d %H:%M:%S")
  echo "$datetime Memory Usage: $memory" >> memory.log
  sleep 3
done`

**Comment by rvdd1962** [comment]
Yup it does, like a hungry dog want to eat more hotdogs :-). Multi-threading issue? Stable diffusion webui doesn't do this with it's memory usage.
![memory usage Fooocus](https://github.com/lllyasviel/Fooocus/assets/149718040/5821af07-2588-415e-b366-c2df9512fdea)

**Comment by yxt132** [comment]
any solution?

**Comment by rvdd1962** [comment]
Still working on it. Got some results but i need to tweak more. Here is the current state:
![test-3](https://github.com/lllyasviel/Fooocus/assets/149718040/fb2734a9-16a8-41e8-a967-ecc67d092f9c)

**Comment by yxt132** [comment]
Any progress?

**Comment by aryan1107** [comment]
yes same issue for me!
